src/ImageSharp.Drawing/Processing/GradientBrush.cs

```diff
@@ -125,12 +125,13 @@ internal abstract class GradientBrushApplicator<TPixel> : BrushApplicator<TPixel
                         var toAsVector = to.Color.ToVector4();
                         float onLocalGradient = (positionOnCompleteGradient - from.Ratio) / (to.Ratio - from.Ratio);
 
-                        // TODO: this should be changeble for different gradienting functions
                         Vector4 result = PorterDuffFunctions.NormalSrcOver(
                             fromAsVector,
                             toAsVector,
                             onLocalGradient);
 
+                        result.W = fromAsVector.W + ((toAsVector.W - fromAsVector.W) * onLocalGradient);
```

Comment:
Can you please explain to me what is happening here? I'm struggling to follow. 

The method you are ultimately calling is:

https://github.com/SixLabors/ImageSharp/blob/41c76515f7e8aeda9a2e7c4a1efa52fd9a3f5ac6/src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.cs#L159

Where `source` and `blend` are the `toAsVector` parameter with the `W` property multiplied by `onLocalGradient`.

Reply:
Sure.
I was not able figure out if there's a better matching PorterDuffFunction available...
I guess you're right and it's just Over() with the right parameters.

Looked for something like that but gave up finally  and added the comment to the PR (see above).
So I'll check that later (hope to get to it in the evening).

Reply:
I'm a noob with Porter-Duff but I've just found the best explanation of the various equations I've ever seen.

https://www.amanithvg.com/docs/tut/007-blending.html

Reply:
Also the old spec docs are waaaaaay better than the new ones.

https://www.w3.org/TR/2009/WD-SVGCompositing-20090430/

Reply:
That's not working as expected.
tried what you proposed here locally now:
```csharp
var fromAsVector = from.Color.ToVector4();
var toWithScaledW = to.Color.ToVector4();
float onLocalGradient = (positionOnCompleteGradient - from.Ratio) / (to.Ratio - from.Ratio);
toWithScaledW.W *= onLocalGradient;

Vector4 result = PorterDuffFunctions.Over(fromAsVector, toWithScaledW, toWithScaledW);

TPixel resultColor = default;
resultColor.FromVector4(result);
return resultColor;
```

but got the following result which is not the correct one:
![GradientsWithTransparencyOnExistingBackground_Rgba32_Blank200x200](https://user-images.githubusercontent.com/2063360/68326603-0e367600-00cc-11ea-8723-0733ea1f70ff.png)

Looks like it's exactly the opposite "direction" of the alpha component than expected.
I checked back with Inkscape, using a red rectangle and a linear gradient on top of that and that agrees with @equinox2k on the expected output.

Have to go for now, will have a look again later.

Reply:
Ok.... Not got a solution but looking at `GradientBrush` class I see an issue that should be fixed. We are not using the passed blender.

Reply:
Ok Now that I've wired up the correct blender I can see the issue more clearly.

Here's how the alpha component of the returned image is computed in `SrcAtop`. 

https://github.com/SixLabors/ImageSharp/blob/41c76515f7e8aeda9a2e7c4a1efa52fd9a3f5ac6/src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.cs#L194

I would say that this is correct since overlaying a semi-transparent pixel atop a background should preserve the background opacity.

So what's the fix? Should we adjust the result alpha component following blending or should we adjust the destination alpha component before?

Reply:
![GradientsWithTransparencyOnExistingBackground_Rgba32_Blank1024x1024](https://user-images.githubusercontent.com/385879/68449817-cdaa3a00-023c-11ea-8ae5-551103302169.png)

@jongleur1983  So, been putting some more thought to it as I figure my way through the code. 

You shouldn't be using a blender when calculating the gradient here. Lerping between the color stops based on the `onLocalGradient` would, imo, be the correct way to do it.

Here's the current output, based upon that approach (Combined with refactoring I'm doing for #1046. 

Now the funny thing.

@equinox2k By my understanding your reference image is incorrect. If you look at the pixel data on the left hand side of my diff you will see a red component value of 245 - Given that the red component of both the background and the source is 255, this should not be possible.

![image](https://user-images.githubusercontent.com/385879/68450207-4fe72e00-023e-11ea-9cde-f07d81d1b585.png)

Here's the image with the second color stop updated to `.1F` instead of `.25F`. This looks pretty good to me.

![GradientsWithTransparencyOnExistingBackground_Rgba32_Blank1024x1024](https://user-images.githubusercontent.com/385879/68451266-2d571400-0242-11ea-8a70-f8ec147de4f6.png)


.

Reply:
The expected result image was just something I knocked up in photoshop quickly to try and show what I expected so likely to have some differences than real output. 

Agree the output looks great “SHIP IT” :)

Reply:
😄 See https://github.com/SixLabors/ImageSharp/pull/1046/commits/ff023b833e3dee44c6bd01b6b17099afc397d435 (_Ignore wrong title in the PR commit, I'm not going to rebase and force push to fix_).

@jongleur1983 If you agree with my findings please let me know and we'll close this PR and wait on mine to be completed.

Reply:
@JimBobSquarePants agreeing with your changes (well, I just looked at the GradientBrush class of that PR, so don't count it as a complete PR review.